### PR TITLE
MigrateMessageCode to manage new case sensitive md5 hash

### PR DESCRIPTION
**Pull request by matteotrubini**
New strings found as per new case sensitive `makeMessageCode()` get a copy of old translation strings.
This ensure no regression in translation quality.

Related: #2

### CODE REVIEW

File: updates/v2.1.0/migrate_message_code.php [hunk 1]
```diff
@@ -11,9 +12,29 @@ class MigrateMessageCode extends Migration
...
+        
         foreach (Message::all() as $message) {
-            $default_message = $message->message_data['x'];
-            $message->code = Message::makeMessageCode($default_message);
+            $message->code_pre_2_1_0 = $message->code;
+            $message->code = $message->code_md5 ?: Message::makeMessageCode($message->message_data[Message::DEFAULT_LOCALE]);
+            $message->save();
+        }
+        
+        if (Schema::hasColumn(self::TABLE_NAME, 'code_md5')) {
```
Comment (mjauvin) [inline updates/v2.1.0/migrate_message_code.php:28]: Where does `code_md5` column comes from?

Comment (matteotrubini) [reply updates/v2.1.0/migrate_message_code.php:28]: it is created by `down()` to preserve md5 hashes on downgrade  (can be useful for manual review).

Comment (mjauvin) [reply updates/v2.1.0/migrate_message_code.php:28]: Oh, I see. So why not use it if it exists instead of re-generate it, fair enough.

**Comment by mjauvin** [comment]
This looks fine at first glance, will review soon.

Was this tested extensively or not?

File: updates/v2.1.0/migrate_message_code.php (hunk 1 above)
Comment (matteotrubini) [reply updates/v2.1.0/migrate_message_code.php:28]: That line is triggered only if `up()` -> `down()` -> `up()`, which is a marginal situation, but I would like to manage it.
I prefer to drop it on `up()` to keep table "clean" and avoid misleading in future.

**Comment by matteotrubini** [comment]
> Was this tested extensively or not?

I made several manual tests over a dev installation based on sqlite - which I expect is more "strict" about table changes.

IMHO users should be **explicitly** alerted to make an export / a backup of their translation before apply this update, not just the usual `!!!`

**Comment by mjauvin** [comment]
@matteotrubini I merged #2 and wanted to merge this one, but there are conflicts that needs to be resolved.

Please let me know when it's done.

**Comment by LukeTowers** [comment]
@matteotrubini @mjauvin whoops, didn't realize that branch was the target for this PR, this PR will need to be resubmitted against main. @mjauvin does having the changes from #2 already merged in break any existing installs or will it cause breakages to happen if this gets merged in after that one is already available?

**Comment by mjauvin** [comment]
not following you here @LukeTowers

**Comment by matteotrubini** [comment]
@LukeTowers I'll reopen an updated PR in hours.
@mjauvin I'll also remove `mb_strtolower()` as discussed in [#2](https://github.com/wintercms/wn-translate-plugin/issues/2#issuecomment-825976983), ok?

**Pull request closed without merging**